spec/openai/client/chat_spec.rb

```diff
@@ -24,6 +26,46 @@
           end
         end
 
+        context "with an invalid function call" do
+          let(:cassette) { "#{model} function call chat".downcase }
+          let(:messages) do
+            [
+              {
+                "role" => "function",
+                # "name" => "function",
+                "content" => "function"
+              }
+            ]
+          end
+          let(:functions) do
+            [
+              {
+                "name" => "function",
+                "description" => "function",
+                "parameters" =>
+                  {
+                    "type" => "object",
+                    "properties" => {
+                      "user" => {
+                        "type" => "string",
+                        "description" => "the full name of the user"
+                      }
+                    }
+                  }
+              }
+            ]
+          end
+
+          it "raises an error containing the reason" do
+            VCR.use_cassette(cassette) do
+              response
+            rescue Faraday::Error => e
+              expect(e.response.dig(:body, "error",
+                                    "message")).to include("Missing parameter 'name'")
+            end
+          end
+        end
```

Comment:
This spec is kind of unrelated but being unable to find this error is what triggered this PR. So including it